data_set/README.md

```diff
@@ -0,0 +1,71 @@
+- `taskset/`: Contains the final set of posts that are going to be used in the user experiment (referenced as _sampled posts_).
+- `web-scrapers/`: Contains the web scrapping scripts to extract the posts information from the websites.
+- `utilities/`: Contains information about Tellina's training set.
+- `clean_stackexchange.py`: Python script that cleans raw post data for one stack exchange website.
+- `clean_stackexchanges.sh`: Bash script that calls `clean_stackexchange.py` for all stack exchange websites.
+- `sample_tasks.ipynb`: Jupyter Notebook that randomly samples _compatible posts_ located in `data-filtered/` to build the taskset.
+- `interactive_filter_stackexchange.py` Interactive python script that filters _candidate posts_ into _compatible posts_.
+
+
+## Data Process
+### 1. Post Acquisition
+The posts is collected through via web-scrapping on five source websites: 
+- [StackOverflow](https://stackoverflow.com/)
+- [SuperUser](https://superuser.com/)
+- [Unix&Linux](https://unix.stackexchange.com/)
+- [CommandLineFu](https://www.commandlinefu.com/)
+- [Bash One-Liners](http://www.bashoneliners.com/)
+
+The web-scrapping scripts for CommandLineFu and Bash One-Liners are located in the `web-scrapers/` folder. For the stack exchange websites (StackOverflow, SuperUser, and Unix&Linux), the posts are obtained using the [Stack Exchange Data Explorer](https://data.stackexchange.com/). See `web-scrapers/` for more details.
+
+The data obtained from this step is stored in the `data-collected` folder. We refer to this data set as _raw posts_.
+
+The _raw posts_ totals 2287 posts (Bash One-Liners has only 287 records for the entire website). We verified that the most popular questions have a score of at least 0.
+
+### 2. Post Cleaning
+This step cleans the data obtained from the previous step into something that can be parsed and automated easily.
+
+For CommandLineFu and Bash One-Liners, the process is minimal: We prepend each line with the base URL of the website using the following commands:
+* `sed -e 's#^#http://www.bashoneliners.com#' data-collected/bashoneliners-top500.csv > bashoneliners-top500-clean.csv`
+* `sed -e 's#^#https://www.commandlinefu.com#' data-collected/commandlinefu-top500.csv > commandlinefu-top500-clean.csv`
+
+For the stack exchange websites, we clean them using the Bash script `clean_stackexchanges.sh`. This script will call `clean_stackexchange.py` for each stack exchange website.
+
+The data obtained from this step is stored in the `data-clean` folder. We refer to this data set as _candidate posts_.
+
+### 3. Post Filtering
+The goal of this step is too filter the _candidate posts_ that are compatible with Tellina and our user experiment.
+
+For CommandLineFu and Bash One-Liners, the process uses various RegExps that are stored in `data-filtered/filters.txt`. Each command is called manually to gradually remove incompatible posts from the _candidate posts_.
```

Comment:
That's a good question. I contemplated putting it in the root of `data_set/` or in the utilities. 

It's a `.txt` because I used a manual process to filter out out-of-scope commands as I encountered them. Adapting each command as I relearned Bash and what regexp I needed to be more efficient. Thus, I didn't spend the time consolidating them into a script since I had gotten what I wanted.